fedimint-server/src/config.rs

```diff
@@ -418,13 +476,25 @@ impl ServerConfig {
         let hbbft_keys = keys[&KeyType::Hbbft].threshold_crypto();
         let epoch_keys = keys[&KeyType::Epoch].threshold_crypto();
 
-        let mut module_cfgs: BTreeMap<String, ServerModuleConfig> = Default::default();
+        let mut module_cfgs: BTreeMap<ModuleInstanceId, ServerModuleConfig> = Default::default();
 
-        for (name, gen) in module_config_gens.iter() {
+        // NOTE: Currently we do not implement user-assisted module-kind to module-instance-id assignment
+        // We assume that user wants one instance of each module that was compiled in. This is how
+        // things were initially, where we consider "module as a code" as "module as an instance at runtime"
+        for (module_instance_id, (_kind, gen)) in module_config_gens.iter().enumerate() {
```

Comment:
So this will have to change if we actually want to support multiple instances per module?

Reply:
Yes. User will have to provide a config of some kind (click it through in the UI) defining `id -> kind` mapping of modules they want.